src/pcap/bosh_resolver.go

```diff
@@ -0,0 +1,208 @@
+package pcap
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"go.uber.org/zap"
+
+	"github.com/cloudfoundry/pcap-release/src/pcap/bosh"
+)
+
+type BoshResolver struct {
+	environment bosh.Environment
+	client      *http.Client
+	uaaURLs     []string
+	agentPort   int
+}
+
+func NewBoshResolver(environment bosh.Environment, agentPort int) (*BoshResolver, error) {
+	resolver := &BoshResolver{environment: environment, agentPort: agentPort} // TODO: get agent port from where?
```

Comment:
I think adding the port to an extended `Environment` that augments the regular BOSH config "Environment" might be the way to go.